dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/config/DubboConfigEarlyRegistrationPostProcessor.java

```diff
@@ -0,0 +1,139 @@
+            this.beanFactory = unwrap(beanFactory);
+            registryConfigEarlyInitializationPostProcessor(this.beanFactory);
+        }
+    }
+
+    @Override
+    public int getOrder() {
+        return HIGHEST_PRECEDENCE;
+    }
+
+    /**
+     * Register DubboConfigEarlyInitializationPostProcessor as BeanPostProcessor manually
+     * before {@link AbstractApplicationContext#registerBeanPostProcessors(ConfigurableListableBeanFactory)
+     * Spring Framework automatically register}
+     */
+    private void registryConfigEarlyInitializationPostProcessor(DefaultListableBeanFactory beanFactory) {
+        if (beanFactory != null) {
+            // Register DubboConfigEarlyInitializationPostProcessor
+            beanFactory.addBeanPostProcessor(configEarlyInitializationPostProcessor);
+            if (logger.isInfoEnabled()) {
+                logger.info("DubboConfigEarlyInitializationPostProcessor has bean registered");
+            }
+        }
+    }
+
+    private DefaultListableBeanFactory unwrap(Object registry) {
+        if (registry instanceof DefaultListableBeanFactory) {
+            return (DefaultListableBeanFactory) registry;
+        }
+        return null;
+    }
+
+    class DubboConfigEarlyInitializationPostProcessor extends GenericBeanPostProcessorAdapter<AbstractConfig> {
+
+        protected void processBeforeInitialization(AbstractConfig config, String beanName) throws BeansException {
+            if (beanFactory == null) {
+                if (logger.isErrorEnabled()) {
+                    logger.error("Current Processor is not running in Spring container, next action will be skipped!");
+                }
+                return;
+            }
+
+            // If CommonAnnotationBeanPostProcessor is already registered,  the method addIntoConfigManager()
+            // will be invoked in Bean life cycle.
+            if (!hasRegisteredCommonAnnotationBeanPostProcessor()) {
+                if (logger.isWarnEnabled()) {
+                    logger.warn("CommonAnnotationBeanPostProcessor is not registered yet, " +
+                            "the method addIntoConfigManager() will be invoked directly");
+                }
+                config.addIntoConfigManager();
+            }
+        }
+
+        /**
+         * {@link DefaultListableBeanFactory} has registered {@link CommonAnnotationBeanPostProcessor} or not?
+         *
+         * @return if registered, return <code>true</code>, or <code>false</code>
+         */
+        private boolean hasRegisteredCommonAnnotationBeanPostProcessor() {
+            for (BeanPostProcessor beanPostProcessor : beanFactory.getBeanPostProcessors()) {
+                if (CommonAnnotationBeanPostProcessor.class.equals(beanPostProcessor.getClass())) {
+                    return true;
+                }
+            }
+            return false;
+        }

```

Comment:
It's better to cache the result if has found CommonAnnotationBeanPostProcessor

Reply:
> It's better to cache the result if has found CommonAnnotationBeanPostProcessor

Thanks for your reply. I can't cache  the result all the time. Here is my reason:
There might be some BeanPostProcessors not instantiated before some Dubbo Config Bean instantiated (which will call this method)，so the  result of `beanFactory.getBeanPostProcessors()` may not be the same when this method is called.     
But, it can be cached after CommonAnnotationBeanPostProcessor instantiated,  I will submit another PR.

Reply:
new PR #9414